src/panel.ts

```diff
@@ -28,8 +32,31 @@ export class CommentPanel extends Panel {
     node.setAttribute('contentEditable', 'true');
     node.classList.add('jc-CommentInput');
     const inputWidget = (this._inputWidget = new Widget({ node }));
-    this.addWidget(inputWidget);
 
+    let filePath: string; 
+    options.labShell.currentChanged.connect((_, args) => {
+      if (args.newValue instanceof DocumentWidget){
+        const docWidget = args.newValue as DocumentWidget;
```

Comment:
If this causes a loop, try logging `docWidget` and seeing what it's changing back-and-forth between

Reply:
I tried doing that but I don't see anything useful. To be more descriptive, it does log as a change and I can see it update the documentwidget but that for some reason causes the page to crash

Reply:
Hmm, that means the infinite looping is happening further down. Try using the performance monitor of your browser to see where it gets stuck.